spotbugs/src/main/java/edu/umd/cs/findbugs/detect/UncallableMethodOfAnonymousClass.java

```diff
@@ -59,6 +59,11 @@ public UncallableMethodOfAnonymousClass(BugReporter bugReporter) {
 
     @Override
     public void visitJavaClass(JavaClass obj) {
+        if (Values.DOTTED_JAVA_LANG_ENUM.equals(obj.getSuperclassName()) ||
+                Values.DOTTED_JAVA_LANG_RECORD.equals(obj.getSuperclassName())) {
+            return;
+        }
+
```

Comment:
The test passes even when I revert this change. I assume that the test case is not enough to reproduce the reported problem.

```console
$ ./gradlew clean spotbugs-tests:test --tests=Issue2120Test
```

Reply:
Thank you for noticing this. I forgot to add the inner classes to the test. (I always do the tests first, run them manually to see them fail, and then I do the fix.)